waspc/examples/todoApp/ext/pages/Task.js

```diff
@@ -2,19 +2,44 @@ import React from 'react'
 import { Link } from 'react-router-dom'
 
 import { useQuery } from '@wasp/queries'
+import { useAction } from '@wasp/actions'
 import updateTaskIsDone from '@wasp/actions/updateTaskIsDone'
 import getTask from '@wasp/queries/getTask.js'
+import getTasks from '@wasp/queries/getTasks.js'
 
 const Todo = (props) => {
   const taskId = parseInt(props.match.params.id)
   const { data: task, isFetching, error } = useQuery(getTask, { id: taskId })
 
+  const updateTaskIsDoneAction = useAction(updateTaskIsDone, {
+    optimisticUpdates: [
+      {
+        getQuery: () => [getTask, { id: taskId }],
+        // This query's cache should should never be emtpy
+        updateQuery: ({ isDone }, oldTask) => ({ ...oldTask, isDone }),
+      },
+      {
+        getQuery: () => [getTasks],
+        updateQuery: (updatedTask, oldTasks) => {
+          if (oldTasks === undefined) {
+            // cache is empty
+            return [updatedTask]
+          } else {
+            return oldTasks.map(task => 
+              task.id === updatedTask.id ? { ...task, ...updatedTask } : task
+            )
+          }
+        },
+      }
+    ]
+  })
+
   if (!task) return <div>Task with id {taskId} does not exist.</div>
   if (error) return <div>Error occurred! {error}</div>
 
   async function toggleIsDone() {
     try {
-      updateTaskIsDone({ id: task.id, isDone: !task.isDone })
+      updateTaskIsDoneAction.mutateAsync({ id: task.id, isDone: !task.isDone })
```

Comment:
@shayneczyzewski said:

----
I am glad you called this out. I don't have a great answer but was taken aback by the function named and assumed we were just using what was already there (but I'm not a React Query guru). I don't think we use the term "mutate" anywhere in user-facing docs, do we? If they are familiar with the API, they will probably grok it, but my eyes did notice that as "something new in Wasp" pretty quickly too.

Reply:
Yes, I've discussed this with @Martinsos on several occasions. I think the correct approach at this point would be renaming/redocumenting everything in terms of "action".

It is a bit unfortunate that React Query uses the term mutation for the same purpose, but iiwii (I think action is more appropriate anyway).

Reply:
Ah yes this is tricky.
I am not sure what to do! What are our options?
1. We have a mix of action and mutate terms -> action coming from Wasp, mutate coming from react query.
2. We rename all "mutate"s in react query API to be "action". But that sounds pretty dirty to me and people will have hard time looking up react query's docs. We could rewrite their docs like blitz does but I am not sure it is worth investing such effort, I like much better just pointing them to existing docs.
3. We drop term "action" in Wasp and call it "mutation". I think "action" is more correct term but what can we do. This is a lot of changes but it will work ok once we do it. Although I will always be slightly unhappy that it is called mutation and not action :D.

Is there any other option that I didn't mention? What do you think?
@sodic you said 
> I think the correct approach at this point would be renaming/redocumenting everything in terms of "action".

Does that mean doing option 2 or option 3?

Reply:
It was option 2, but now I don't like it anymore :upside_down_face: 

There is a third option, though. I could try to implement optimistic updates without using React Query. From what I've seen so far, it should be fairly straightforward. We would be losing the benefits of `useAction` though, but it would also reduce complecity.

Reply:
Uff that sounds like can of worms we might not want to do right now? It might get complicated + we need to write docs then.
So we have 4 options now. What do you think about each of them?

Reply:
Ok, I took a fifth option. I restricted the API and "covered up" React Query. This means we won't be offering the additional benefits of their hooks, but our API and naming is more consistent. You can check the code for details.